src/mkcalworker.cpp

```diff
@@ -568,9 +568,11 @@ bool mKCalWorker::saveCollections(QList<QOrganizerCollection> *collections,
         int index = 0;
         for (QList<QOrganizerCollection>::Iterator it = collections->begin();
              it != collections->end(); ++it, ++index) {
+            mKCal::Notebook::Ptr nb;
             if (it->id().isNull()) {
-                mKCal::Notebook::Ptr nb(new mKCal::Notebook);
+                nb = mKCal::Notebook::Ptr(new mKCal::Notebook);
                 updateNotebook(nb, *it);
+
                 if (!mStorage->addNotebook(nb)) {
```

Comment:
Actually `setDefaultNotebook(nb)` is adding if not present (and updating if already) to avoid two database operation, first an addition and then an update for the default flag.

I would suggest to call directly `setDefaultNotebook()` here in case of changing default. Before the `if (id().isNull())`, store in a bool if the default is changing. Then, do something like :
```
if (!(isDefault ? mStorage->setDefaultNotebook(nb) : mStorage->addNotebook(nb)))
```
And the same in the update case. After the if, the `mDefaultNotebookUid` value is updated if necessary and signal is emitted.